src/app/components/quiz.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { useState } from "react";
+import clsx from "clsx";
+
+export interface QuizParams {
+  className?: string;
+  question: string;
+  hint?: string;
+  answers: string[];
+  correctAnswer: number;
+  hintAnswers?: string[];
+  onSelect?: (selection: number, isDone: boolean) => void;
+  // If enabled, the user only has a single try. After selection, it will show the solution to the user.
+  showCorrectAnswer?: boolean;
+}
+
+/**
+ * A quiz component.
+ * @param quiz The quiz parameters
+ */
+export default function Quiz(quiz: QuizParams) {
+  const [selection, setSelection] = useState<number>();
+
+  function onClick(index: number) {
+    if (quiz.showCorrectAnswer && selection != undefined) return;
+    setSelection(index);
+    const isDone = quiz.showCorrectAnswer
+      ? index != undefined
+      : index == quiz.correctAnswer;
+    console.log("Quiz", isDone, quiz.showCorrectAnswer, selection);
+    quiz.onSelect?.(index, isDone);
+  }
+
+  const selectionClass = (index: number) => {
+    if (
+      quiz.showCorrectAnswer &&
+      selection != undefined &&
+      index == quiz.correctAnswer
+    ) {
+      return "bg-lime-500";
+    } else {
+      return selection == index
+        ? index == quiz.correctAnswer
+          ? "bg-lime-500"
+          : "bg-red-500"
+        : "";
+    }
+  };
```

Comment:
be consistent. 
decide if you want to use function or const as function declaration.

Reply:
fair point 😅